File: Mathematics/Armstrong_no/C/armstrong_no.c
/**
 * C program to check Armstrong number
 */
#include <stdio.h>
#include <math.h>

int main()
{
    int originalNum, num, lastDigit, digits, sum;

    /* Input number from user */
    printf("Enter any number to check Armstrong number: ");
    scanf("%d", &num);

    sum = 0;

    /* Copy the value of num for processing */
    originalNum = num;

    /* Find total digits in num */
    digits = (int) log10(num) + 1;

    /* Calculate sum of power of digits */
    while(num > 0)
    {
        /* Extract the last digit */
        lastDigit = num % 10;

        /* Compute sum of power of last digit */
        sum = sum + round(pow(lastDigit, digits));

        /* Remove the last digit */
        num = num / 10;
    }

    /* Check for Armstrong number */
    if(originalNum == sum)
    {
        printf("%d is ARMSTRONG NUMBER", originalNum);
    }
    else
    {
        printf("%d is NOT ARMSTRONG NUMBER", originalNum);
    }

    return 0;
}

File: Mathematics/Armstrong_no/csharp/armstrong_no.cs
using System;

static class Program
{
    public static void Main()
    {
        int origNum, num, rem, sum = 0;

        Console.Write("Enter a positive integer: ");
        origNum = int.Parse(Console.ReadLine());
        num = origNum;

        while(num != 0)
        {
            rem = num % 10;
            sum += rem * rem * rem;
            num /= 10;
        }

        if (sum == origNum)
            Console.WriteLine($"{origNum} is an Armstrong number.");
        else
            Console.WriteLine($"{origNum} is not an Armstrong number.");
    }
}